gravitee-apim-gateway/gravitee-apim-gateway-reactor/src/main/java/io/gravitee/gateway/reactor/handler/impl/DefaultReactorHandlerRegistry.java

```diff
@@ -203,52 +200,55 @@ private Class<? extends Acceptor<?>> resolve(Class<? extends Acceptor> acceptor)
         );
     }
 
-    private void addAcceptors(List<? extends Acceptor<?>> newAcceptors) {
+    private void registerAcceptors(List<? extends Acceptor<?>> newAcceptors) {
         if (!newAcceptors.isEmpty()) {
             synchronized (this) {
                 newAcceptors.forEach(acceptor -> {
                     Class<? extends Acceptor<?>> acceptorType = resolve(acceptor.getClass());
-                    List registeredAcceptors = acceptors.get(acceptorType);
-                    if (registeredAcceptors == null) {
-                        registeredAcceptors = new ArrayList<>();
+                    if (acceptorType != null) {
+                        acceptors.compute(
+                            acceptorType,
+                            (k, v) -> {
+                                if (v == null) {
+                                    v = new ArrayList<>();
+                                }
+                                v.add(acceptor);
+                                // Sort list based on Acceptor comparable
+                                v.sort(null);
```

Comment:
Collections.sort() ?

Reply:
Cannot be used because of generic, but Collections.sort(X) does x.sort(null)

Reply:
![image](https://github.com/gravitee-io/gravitee-api-management/assets/4324887/e464f737-9032-46c2-a3e5-873bb6304715)